### PR TITLE
Flp 670 sonarcloud code rating improvements

### PR DESCRIPTION
**Pull request by tom-gough**

### CODE REVIEW

File: src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfPrice/ChangeOfPriceEmployerController.cs [hunk 1]
```diff
@@ -153,9 +154,9 @@ public async Task<IActionResult> CancelPriceChange(string employerAccountId, str
     public async Task<IActionResult> ApproveOrRejectPriceChangePage(string employerAccountId, string apprenticeshipHashedId, string ApproveChanges, string rejectReason)
     {
         var apprenticeshipKey = await _apprenticeshipService.GetApprenticeshipKey(apprenticeshipHashedId);
-        if (apprenticeshipKey == default)
+        if (apprenticeshipKey == Guid.Empty)
         {
-            _logger.LogWarning($"Apprenticeship key not found for apprenticeship with hashed id {apprenticeshipHashedId}");
+            _logger.LogWarning("Apprenticeship key not found for apprenticeship with supplied apprenticeshipHashedId");
```
Comment (MikeV-TalentConsulting) [inline src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfPrice/ChangeOfPriceEmployerController.cs:159]: Where losing the hashId in the logs here

File: src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfPrice/ChangeOfPriceProviderController.cs [hunk 2]
```diff
@@ -194,9 +193,9 @@ public async Task<IActionResult> CancelPriceChange(long ukprn, string apprentice
         }
 
         var apprenticeshipKey = await _apprenticeshipService.GetApprenticeshipKey(apprenticeshipHashedId);
-        if (apprenticeshipKey == default)
+        if (apprenticeshipKey == Guid.Empty)
         {
-            _logger.LogWarning($"Apprenticeship key not found for apprenticeship with hashed id {apprenticeshipHashedId}");
+            _logger.LogWarning("Apprenticeship key not found for apprenticeship with supplied apprenticeshipHashedId");
```
Comment (MikeV-TalentConsulting) [inline src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfPrice/ChangeOfPriceProviderController.cs:198]: Losing HashId in logs here

File: src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfStartDate/ChangeOfStartDateEmployerController.cs [hunk 3]
```diff
@@ -70,9 +70,9 @@ public async Task<IActionResult> ViewPendingChangePage(string employerAccountId,
     public async Task<IActionResult> ApproveOrRejectStartDateChange(string employerAccountId, string apprenticeshipHashedId, string approveChanges, string rejectReason)
     {
         var apprenticeshipKey = await _apprenticeshipService.GetApprenticeshipKey(apprenticeshipHashedId);
-        if (apprenticeshipKey == default)
+        if (apprenticeshipKey == Guid.Empty)
         {
-            _logger.LogWarning($"Apprenticeship key not found for apprenticeship with hashed id {apprenticeshipHashedId}");
+            _logger.LogWarning("Apprenticeship key not found for apprenticeship with supplied apprenticeshipHashedId");
```
Comment (MikeV-TalentConsulting) [inline src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfStartDate/ChangeOfStartDateEmployerController.cs:75]: Losing hashId here

File: src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfStartDate/ChangeOfStartDateProviderController.cs [hunk 4]
```diff
@@ -177,9 +178,9 @@ public async Task<IActionResult> CancelStartDateChange(long ukprn, string appren
 		}
 
 		var apprenticeshipKey = await _apprenticeshipService.GetApprenticeshipKey(apprenticeshipHashedId);
-		if (apprenticeshipKey == default)
+		if (apprenticeshipKey == Guid.Empty)
 		{
-			_logger.LogWarning("Apprenticeship key not found for apprenticeship with hashed id {apprenticeshipHashedId}", apprenticeshipHashedId);
+			_logger.LogWarning("Apprenticeship key not found for apprenticeship with supplied apprenticeshipHashedId");
```
Comment (MikeV-TalentConsulting) [inline src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfStartDate/ChangeOfStartDateProviderController.cs:183]: Losing hashId here

File: src/SFA.DAS.Apprenticeships.Web/Identity/Authentication/EmployerAccountPostAuthenticationClaimsHandler.cs [hunk 5]
```diff
@@ -22,14 +22,14 @@ public EmployerAccountPostAuthenticationClaimsHandler(IEmployerAccountService ac
     public async Task<IEnumerable<Claim>> GetClaims(TokenValidatedContext tokenValidatedContext)
     {
         var claims = new List<Claim>();
-        var userId = tokenValidatedContext.Principal.Claims
+        var userId = tokenValidatedContext.Principal?.Claims
```
Comment (MikeV-TalentConsulting) [inline src/SFA.DAS.Apprenticeships.Web/Identity/Authentication/EmployerAccountPostAuthenticationClaimsHandler.cs:25]: We allow the collection of a null value here then throw on line 32 if its null. Would be better to throw here if Principle is null.

**Review by MikeV-TalentConsulting — APPROVED** [review]

File: src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfPrice/ChangeOfPriceEmployerController.cs (hunk 1 above)
Comment (tom-gough) [reply src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfPrice/ChangeOfPriceEmployerController.cs:159]: That was intentional, logs whatever the user puts in the url, could be a data protection issue

File: src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfPrice/ChangeOfPriceProviderController.cs (hunk 2 above)
Comment (tom-gough) [reply src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfPrice/ChangeOfPriceProviderController.cs:198]: Intentional as above

File: src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfStartDate/ChangeOfStartDateEmployerController.cs (hunk 3 above)
Comment (tom-gough) [reply src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfStartDate/ChangeOfStartDateEmployerController.cs:75]: Intentional as above

File: src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfStartDate/ChangeOfStartDateProviderController.cs (hunk 4 above)
Comment (tom-gough) [reply src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfStartDate/ChangeOfStartDateProviderController.cs:183]: Intentional as above

**Review by MikeV-TalentConsulting — APPROVED** [review]

**Pull request merged**